app/src/main/java/org/wikipedia/readinglist/ReadingListsFragment.kt

```diff
@@ -532,6 +544,48 @@ class ReadingListsFragment : Fragment(), SortReadingListsDialog.Callback, Readin
         }
     }
 
+    private fun maybeShowImportReadingListsDialog() {
+        val encodedJson = Prefs.importReadingListsData
+        if (!Prefs.importReadingListsDialogShown && !encodedJson.isNullOrEmpty()) {
+            binding.swipeRefreshLayout.isRefreshing = true
+            lifecycleScope.launch(CoroutineExceptionHandler { _, throwable ->
+                L.e(throwable)
+                FeedbackUtil.showError(requireActivity(), throwable)
+                binding.swipeRefreshLayout.isRefreshing = false
+            }) {
+                withContext(Dispatchers.Main) {
+                    val readingList = ReadingListsImportHelper.importReadingLists(requireContext(), encodedJson)
+                    val dialogView = ReadingListImportDialogView(requireContext())
+                    dialogView.setReadingList(readingList)
+                    ReadingListsFunnel().logReceivePreview(readingList)
+                    AlertDialog.Builder(requireContext())
+                        .setView(dialogView)
+                        .setPositiveButton(R.string.shareable_reading_lists_import_dialog_confirm) { _, _ ->
+                            ReadingListsFunnel().logReceiveFinish(readingList)
+                            importReadingListAndRefresh(readingList)
+                        }
+                        .setNegativeButton(R.string.shareable_reading_lists_import_dialog_cancel) { _, _ ->
+                            ReadingListsFunnel().logReceiveCancel(readingList)
+                        }
+                            .setOnDismissListener {
+                                ReadingListsSurveyHelper.activateSurvey()
+                                ReadingListsSurveyHelper.maybeShowSurvey(requireActivity())
+                            }
+                            .show()
+                    Prefs.importReadingListsDialogShown = true
+                    binding.swipeRefreshLayout.isRefreshing = false
+                }
+            }
+        }
+    }
+
+    private fun importReadingListAndRefresh(readingList: ReadingList) {
```

Comment:
Will it be possible to move this or even the entire logic of showing the dialog to the helper class?

Reply:
Perhaps in a follow-up. At the moment it seems a bit too tightly bound to the Fragment logic.